terrapower/physics/neutronics/dragon/dragonFactory.py

```diff
@@ -0,0 +1,48 @@
+"""Factory for building DRAGON related objects and their subclasses"""
+
+
+class DragonFactory:
+    """
+    Build objects based on registration.
+
+    Enables easy customization of the chain of objects needed without excessive subclassing
+    """
+
+    def __init__(self):
+        self._writers = {}
+        self._executers = {}
+        self._runners = {}
+        self._key = None
+
+    def setKey(self, key):
+        """Apply a key to this factory to choose classes with."""
+        self._key = key
+
+    def copyEntriesToKey(self, newKey):
```

Comment:
it's called in client code in the hallam application so I put it there only because I did indeed need it. Happy to write more docs about why it's there though, so good comment. 

I was trying to explore the concept of the Abstract Factory design pattern. maybe I did poorly.